services/QuillLMS/app/models/subscription.rb

```diff
@@ -80,15 +83,26 @@ class Subscription < ApplicationRecord
   SCHOOL_FIRST_PURCHASE_PRICE = SCHOOL_RENEWAL_PRICE
   TEACHER_PRICE = 8000
   ALL_PRICES = [TEACHER_PRICE, SCHOOL_RENEWAL_PRICE]
-  PAYMENT_METHODS = ['Invoice', 'Credit Card', 'Premium Credit']
+  PAYMENT_METHODS = [
+    INVOICE_PAYMENT_METHOD = 'Invoice',
+    CREDIT_CARD_PAYMENT_METHOD = 'Credit Card',
+    PREMIUM_CREDIT_PAYMENT_METHOD = 'Premium Credit'
+  ]
+
   ALL_TYPES = OFFICIAL_FREE_TYPES.dup.concat(OFFICIAL_PAID_TYPES)
 
+  validates :stripe_invoice_id, allow_blank: true, format: { with: /\Ain_[0-9a-zA-Z]*\z/ }
+
   scope :active, -> { where(de_activated_date: nil).where("expiration > ?", Date.today).order(expiration: :asc) }
 
   def is_trial?
     account_type && TRIAL_TYPES.include?(account_type)
   end
 
+  def expired?
+    expiration <= Date.today
```

Comment:
We might want to add this cop and move away from `Date.today` (in favor of Date.current or `Time.now.utc.to_date`), just to ensure the timezone is utc.

https://www.rubydoc.info/gems/rubocop/0.41.2/RuboCop/Cop/Rails/Date

Reply:
I will update this instance and add then add the Rubocop setting in a future PR since it will require a new gem: rubocop-rails.